src/Components/ResourcePicker/ResourcePicker.tsx

```diff
@@ -0,0 +1,649 @@
+                    ((resourceType ===
+                        AzureResourceTypes.StorageBlobContainer ||
+                        resourceType === AzureResourceTypes.StorageAccount) &&
+                        ValidContainerHostSuffixes.some(
+                            (suffix) =>
+                                newVal.endsWith(suffix) ||
+                                newVal.endsWith(suffix + '/')
+                        ))
+                ) {
+                    newVal = 'https://' + newVal;
+                }
+            }
+            return newVal;
+        } else {
+            return value;
+        }
+    }, []);
+
+    const isValidUrlStr = useCallback((urlStr: string) => {
+        try {
+            let endsWithValidSuffix = true;
+            switch (resourceType) {
+                case AzureResourceTypes.DigitalTwinInstance:
+                    endsWithValidSuffix = ValidAdtHostSuffixes.some((suffix) =>
+                        new URL(urlStr).hostname.endsWith(suffix)
+                    );
+                    break;
+                case AzureResourceTypes.StorageAccount:
+                case AzureResourceTypes.StorageBlobContainer:
+                    endsWithValidSuffix = ValidContainerHostSuffixes.some(
+                        (suffix) => new URL(urlStr).hostname.endsWith(suffix)
+                    );
+                    break;
+            }
+            return (
+                urlStr && urlStr.startsWith('https://') && endsWithValidSuffix
+            );
+        } catch (error) {
+            return false;
+        }
+    }, []);
+
+    const inputError = useMemo(() => {
+        if (
+            selectedOption &&
+            displayField === AzureResourceDisplayFields.url &&
+            !isValidUrlStr(selectedOption.text)
+        ) {
+            switch (resourceType) {
+                case AzureResourceTypes.DigitalTwinInstance:
+                    return t('resourcesPicker.errors.invalidEnvironmentUrl');
+                case AzureResourceTypes.StorageAccount:
+                    return t('resourcesPicker.errors.invalidStorageAccountUrl');
+                case AzureResourceTypes.StorageBlobContainer:
+                    return t('resourcesPicker.errors.invalidContainerUrl');
+                default:
+                    return undefined;
+            }
+        }
+    }, [selectedOption, resourceType, isValidUrlStr, t]);
+
+    const isValuesEqual = (value1: string, value2: string) => {
+        if (displayField === AzureResourceDisplayFields.url) {
+            if (value1?.endsWith('/')) {
+                value1 = value1.slice(0, -1);
+            }
+            if (value2?.endsWith('/')) {
+                value2 = value2.slice(0, -1);
+            }
+            return value1 === value2;
+        } else {
+            return value1 === value2;
+        }
+    };
+
+    const handleOnChange = useCallback(
+        (option, value) => {
+            if (option) {
```

Comment:
can we add some comments to this `onChange` function? Like why would we ever not have an option? And if we don't how can we make the assumption to parse the text?

Reply:
when the fluent combobox's allowfreeform prop is enabled, when onchange two parameters are passed: option and value. Option is referring to an existing dropdown option whereas value is a newly entered freeform value.